main/samba/src/EBox/Samba/User.pm

```diff
@@ -436,6 +436,7 @@ sub create
     my ($class, %args) = @_;
 
     # Check for required arguments.
+    throw EBox::Exceptions::MissingArgument('name') unless ($args{name});
```

Comment:
Good observation, the problem is that I debugged this from a scripts which does not use given name and surname. name is used whcih this two are not provided. I will change this